### ISSUE TITLE
About masks prompt

### ISSUE THREAD

**Issue by wu2233** [rafiibnsultan/GeoSAM#7]
Hello, I noticed that you used the `mask_predictions` from the `mask_decoder` as the `masks` prompt for the `prompt_encoder`. I tried to use a binary image (a prediction result generated by another segmentation model) as the `masks` prompt, but I failed. I would like to know what the requirements are for the `masks` prompt. I observed that it seems to only accept dense embeddings generated from tensors of shape (1, 1, 256, 256) that result in a shape of (1, 256, 64, 64). If you could answer my question, I would be very grateful.

**Comment by rafiibnsultan** [comment]
Hi.
I remember a binary image as the mask directly will not work. You need a specialized soft mask or say logits (each pixel with a probability value) that work as the mask_prompt.

If you check the prediction() in SAM's code you can see:

masks, iou_predictions, low_res_masks = self.predict_torch(
            coords_torch,
            labels_torch,
            box_torch,
            mask_input_torch,
            embeddings_torch,
            multimask_output,
            return_logits=return_logits,
        )

        masks_np = masks[0].detach().cpu().numpy()
        iou_predictions_np = iou_predictions[0].detach().cpu().numpy()
        low_res_masks_np = low_res_masks[0].detach().cpu().numpy()
        return masks_np, iou_predictions_np, #low_res_masks_np
        
This low_res_masks_np is what we used as the mask input which is not a binary mask but you can say a low-quality mask with logits, not yet thresholded yet.
Check here the difference:
![image](https://github.com/user-attachments/assets/9cfe7870-3972-42d1-b46f-3406ad1186cf)

So, if you want to use another model's output you need the mask before they use a threshold the logit values into binary values, in the exact shape (256, 256).


 Let me know if you have any more questions.

**Comment by wu2233** [comment]
Thank you very much for your response. I have been so busy recently that I haven't been able to think and try. Regarding the binary masks generated by other models that I mentioned, you can consider that I only have the mask images, hence I cannot obtain their logit values. Do you have any suggestions for me on how to make use of these inaccurate binary images?

**Comment by rafiibnsultan** [comment]
No, I don't think you can do anything if you have binary masks already. If you don't have access to the code of the other model it would be hard to utilize them as dense prompts.

**Comment by rafiibnsultan** [comment]
Hi,

Since you work in a related domain, I was wondering if you would be interested in reviewing this current work as a paper that we plan to submit to a relevant journal. If you are open to this, I can include you as a potential reviewer. For this, I would need your contact information or a link to your Google Scholar profile.

If you're willing to be a potential reviewer, please feel free to reach out to me via email at: hm4013@wayne.edu.

**Comment by wu2233** [comment]
Thank you again for your assistance. I am not in the field of remote sensing; I am just a newcomer in the area of defect segmentation and would not be qualified to be a potential reviewer

**Comment by wu2233** [comment]
Hello, I've encountered a new issue. When running your train_with_text.py, I received an error message stating "Sizes of tensors must match except in dimension 1. Expected size 1 but got size 64 for tensor number 1 in the list". It seems that the prompt_encoder of SAM does not accept tensors of shape (1,1,256). I'm not sure if I've missed something. Looking forward to your reply

**Comment by rafiibnsultan** [comment]
Hi sorry for the late reply.

Can you elaborate on the issue?
It looks like a simple mismatching issue to me.
